distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Widget_Helpers.enso

```diff
@@ -82,52 +83,47 @@ make_column_name_vector_selector table display=Display.Always =
 
 ## PRIVATE
    Make a column reference by name selector.
-make_column_ref_by_name_selector : Table -> Display -> Widget
-make_column_ref_by_name_selector table display=Display.Always =
-    col_names_options = table.column_names.map (name -> Option name "(Column_Ref.Name "+name.pretty+")")
-    expression_option = Option "<Expression>" "(Column_Ref.Expression '["+table.column_names.first+"]')"
-    Single_Choice values=(col_names_options+[expression_option]) display=display
-
-## PRIVATE
-   Same as `make_column_ref_by_name_selector` but also highlights an option for
-   custom text.
-make_column_ref_or_text_value_selector : Table -> Display -> Widget
-make_column_ref_or_text_value_selector table display=Display.Always =
-    col_names_options = table.column_names.map (name -> Option name "(Column_Ref.Name "+name.pretty+")")
-    custom_text_option = Option "'custom text'" "''"
-    expression_option = Option "<Expression>" "(Column_Ref.Expression '["+table.column_names.first+"]')"
-    Single_Choice values=(col_names_options+[custom_text_option, expression_option]) display=display
+make_column_ref_by_name_selector : Table -> Display -> Boolean -> Boolean -> Boolean -> Boolean -> Widget
+make_column_ref_by_name_selector table display:Display=Display.Always add_text:Boolean=False add_regex:Boolean=False add_number:Boolean=False add_boolean:Boolean=False =
+    text = if add_text then [Option "<Text Value>" "''"] else []
+    regex = if add_regex then [Option "<Regular Expression>" "('').to_regex"] else []
+    number = if add_number then [Option "<Number Value>" "0"] else []
+    boolean = if add_boolean then [Option "<True/False>" "True"] else []
+    expression = if table.is_nothing.not then [Option "<Expression>" "(Column_Ref.Expression '["+table.column_names.first+"]')"] else []
+    col_names = if table.is_nothing then [] else table.column_names.map (name -> Option name "(Column_Ref.Name "+name.pretty+")")
```

Comment:
very very minor nitpick: why is the negeteness of these 2 expressions flipped, I don't see any reason why they couldn't be consistent
```suggestion
    expression = if table.is_nothing.not then [Option "<Expression>" "(Column_Ref.Expression '["+table.column_names.first+"]')"] else []
    col_names = if table.is_nothing.not then table.column_names.map (name -> Option name "(Column_Ref.Name "+name.pretty+")") else []
```